machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -0,0 +1,255 @@
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of the machine config node.
+	// +required
+	Spec MachineConfigNodeSpec `json:"spec"`
+
+	// status describes the last observed state of this machine config node.
+	// +optional
+	Status MachineConfigNodeStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigNodeList describes all of the MachinesStates on the system
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type MachineConfigNodeList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard list metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata"`
+
+	// items contains a collection of MachineConfigNode resources.
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	Items []MachineConfigNode `json:"items"`
+}
+
+// MCOObjectReference holds information about an object the MCO either owns
+// or modifies in some way
+type MCOObjectReference struct {
+	// name is the name of the object being referenced. For example, this can represent a machine
+	// config pool or node name.
+	// Must be a lowercase RFC-1123 subdomain name (https://tools.ietf.org/html/rfc1123) consisting
+	// of only lowercase alphanumeric characters, hyphens (-), and periods (.), and must start and end
+	// with an alphanumeric character, and be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+}
+
+// MachineConfigNodeSpec describes the MachineConfigNode we are managing.
+type MachineConfigNodeSpec struct {
+	// node contains a reference to the node for this machine config node.
+	// +required
+	Node MCOObjectReference `json:"node"`
+
+	// pool contains a reference to the machine config pool that this machine config node's
+	// referenced node belongs to.
+	// +required
+	Pool MCOObjectReference `json:"pool"`
+
+	// configVersion holds the desired config version for the node targeted by this machine config node resource.
+	// The desired version represents the machine config the node will attempt to update to and gets set before the machine config operator validates
+	// the new machine config against the current machine config.
+	// +required
+	ConfigVersion MachineConfigNodeSpecMachineConfigVersion `json:"configVersion"`
+}
+
+// MachineConfigNodeStatus holds the reported information on a particular machine config node.
+type MachineConfigNodeStatus struct {
+	// conditions represent the observations of a machine config node's current state.
```

Comment:
We typically list off the valid conditions types, can we get this godoc updated please